<div style="background:linear-gradient(135deg,#2e1065 0%,#6d28d9 55%,#8b5cf6 100%);border-radius:18px;padding:34px 30px;color:#fff;font-family:Inter,Segoe UI,sans-serif">
  <div style="font-size:12px;letter-spacing:3px;color:#ddd6fe;font-weight:700;text-transform:uppercase">Chapter 45 · Solutions</div>
  <div style="font-size:34px;font-weight:900;line-height:1.1;margin:10px 0 6px">Moments &amp; Inequalities, Worked Answers ✅</div>
  <div style="font-size:15px;color:#ede9fe;max-width:720px;line-height:1.6">Full solutions to the five challenges. Try them yourself first, then compare.</div>
  <div style="margin-top:16px;font-size:13px;color:#c4b5fd">Statistics, Data Science and AI: A Visual Handbook · John Fisher · 2026</div>
</div>

## ⚙️ Setup

In [1]:
import numpy as np
from scipy import stats
rng = np.random.default_rng(451)
print("ready")

ready


<div style="background:#f5f3ff;border-left:5px solid #7c3aed;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#7c3aed;letter-spacing:1px">CHALLENGE 1 · FOUR MOMENTS</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Describe a sample</div>
<div style="color:#4a5578;margin-top:6px">Generate 100,000 draws from a Poisson(3) and report its mean, variance, skewness, and kurtosis.</div>
</div>

In [2]:
X = rng.poisson(3, size=100_000)
print(f"mean = {X.mean():.3f} (true 3), var = {X.var():.3f} (true 3)")
print(f"skew = {stats.skew(X):.3f} (true 1/sqrt(3)={1/np.sqrt(3):.3f}), kurtosis = {stats.kurtosis(X):.3f} (true 1/3)")

mean = 2.997 (true 3), var = 3.028 (true 3)
skew = 0.604 (true 1/sqrt(3)=0.577), kurtosis = 0.369 (true 1/3)


**Answer:** for Poisson(&#955;=3), mean = variance = **3** (its signature), skewness = 1/&#8730;&#955; &approx; **0.577**, and excess kurtosis = 1/&#955; &approx; **0.333**. The moments match the known Poisson formulas.

<div style="background:#f5f3ff;border-left:5px solid #7c3aed;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#6d28d9;letter-spacing:1px">CHALLENGE 2 · MGF</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Recover the mean</div>
<div style="color:#4a5578;margin-top:6px">For X ~ Normal(5, 2), the MGF is exp(mu*t + sigma^2 t^2 / 2). Recover the mean numerically by differentiating the empirical MGF at 0.</div>
</div>

In [3]:
X = rng.normal(5, 2, size=500_000)
M = lambda t: np.mean(np.exp(t*X)); h=1e-3
mean_mgf = (M(h)-M(-h))/(2*h)
print(f"M'(0) = {mean_mgf:.3f}  (true mean 5)")

M'(0) = 4.996  (true mean 5)


**Answer:** the numerical derivative M&#8242;(0) &approx; **5**, the mean. The normal MGF exp(&#956;t + &#963;&#178;t&#178;/2) differentiates at 0 to &#956;, and the empirical estimate confirms it.

<div style="background:#f5f3ff;border-left:5px solid #7c3aed;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#0d9488;letter-spacing:1px">CHALLENGE 3 · CHEBYSHEV</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">A distribution-free bound</div>
<div style="color:#4a5578;margin-top:6px">Test scores have mean 70 and sd 8 (shape unknown). Use Chebyshev to bound the fraction of scores outside 54 to 86.</div>
</div>

In [4]:
# 54 to 86 is mu +/- 2 sd; Chebyshev: P(|X-mu| >= k sd) <= 1/k^2
k = 2
print(f"Chebyshev: P(outside mu +/- {k} sd) <= 1/{k}^2 = {1/k**2:.2f}")
print("So at most 25% of scores fall outside 54-86, whatever the shape.")

Chebyshev: P(outside mu +/- 2 sd) <= 1/2^2 = 0.25
So at most 25% of scores fall outside 54-86, whatever the shape.


**Answer:** 54 and 86 are &#956; &plusmn; 2&#963;, so Chebyshev guarantees at most 1/2&#178; = **25%** lie outside, hence at least 75% inside, for *any* distribution. (If it were normal, the real figure would be about 95% inside, Chebyshev is a worst-case bound.)

<div style="background:#f5f3ff;border-left:5px solid #7c3aed;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#db2777;letter-spacing:1px">CHALLENGE 4 · MARKOV</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">Nonnegative tail</div>
<div style="color:#4a5578;margin-top:6px">Website response times are nonnegative with mean 200 ms. Use Markov's inequality to bound the probability a response takes at least 1000 ms.</div>
</div>

In [5]:
mean, a = 200, 1000
print(f"Markov: P(X >= {a}) <= E[X]/a = {mean}/{a} = {mean/a:.2f}")

Markov: P(X >= 1000) <= E[X]/a = 200/1000 = 0.20


**Answer:** Markov gives P(X &#8805; 1000) &#8804; 200/1000 = **0.20**. With nothing but the mean, we can still cap the chance of a slow response at 20%. Markov is crude but assumption-free, it needs only that X &#8805; 0 and a known mean.

<div style="background:#f5f3ff;border-left:5px solid #7c3aed;border-radius:10px;padding:16px 20px;font-family:Inter,sans-serif">
<span style="font-size:13px;font-weight:800;color:#5b21b6;letter-spacing:1px">CHALLENGE 5 · SAMPLE COMPLEXITY</span>
<div style="font-size:21px;font-weight:800;color:#1a2138;margin-top:4px">How much data?</div>
<div style="color:#4a5578;margin-top:6px">Using Hoeffding, find how many test examples you need to estimate an accuracy within +/- 0.03 with 99% confidence.</div>
</div>

In [6]:
eps, delta = 0.03, 0.01
n = int(np.ceil(np.log(2/delta)/(2*eps**2)))
print(f"n >= ln(2/{delta})/(2*{eps}^2) = {n:,} examples")

n >= ln(2/0.01)/(2*0.03^2) = 2,944 examples


**Answer:** n &#8805; ln(2/0.01)/(2&middot;0.03&#178;) &approx; **2,944** examples. Hoeffding turns a desired precision and confidence into a concrete sample size, the kind of guarantee that underlies PAC learning and honest model evaluation.

---
<div style="background:#ffffff;border:1px solid #e6e9f2;border-radius:16px;padding:22px 26px;font-family:Inter,sans-serif">
<div style="font-size:19px;font-weight:800;color:#1a2138">🎉 Nicely done</div>
<div style="color:#4a5578;line-height:1.8;margin-top:8px">You computed moments, recovered a mean from an MGF, applied Chebyshev and Markov for distribution-free tail bounds, and used Hoeffding to size a test set. Moments describe shape; inequalities bound the tails with minimal assumptions.</div>
</div>

<div style="text-align:center;color:#8b94b3;font-size:12px;margin-top:16px">Statistics, Data Science and AI: A Visual Handbook · © 2026 John Fisher</div>